### PR TITLE
button styling fixed

### LINKED ISSUE #3055: Make button styles consistent in Expensify.cash
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
Button style attributes should remain consistent across all Expensify.cash platforms (including iOS and Android). More specifically, buttons should behave like so:

- They are sticky to the bottom of the page/view
- They won't enable unless the user has completed an action (e.g., typed in a text field, selected a dropdown value, etc.) 
   - When the button is disabled it should have a reduced opacity and we should eliminate mouse/touch events for the disabled states. 
- They have a white margin to the sides and below the button to prevent the look of overlapping content (see image1 below)

## Actual Result:
Button styles are inconsistent right now. Here's a few examples:

image1:
image

image2:
image

## Buttons to modify:

1. Settings > Profile `Save` button 
2. Settings > Profile > `Add Phone Number`/`Add Email` > `Send Validation` button 
2. Settings > Change Password `Save` button 
3. Settings > Payments `Add PayPal Account` button
4. `+` FAB > New Group > Select 1 or more people > `Create Group` button 
5. `+` FAB > Request Money > Enter Amount and Press `Next` > Select an attendee > Enter a comment > Request `$XX button`.
6. `+` FAB > Split Bill > Enter Amount and Press `Next` > Select 1 or more people > `Next` button 
7. `+` FAB > Request Money/Split Bills > Press the currency symbol > Search for a currency using the search bar to focus the keyboard > Select a currency > Confirm button.

## Platform:
Where is this issue occurring?

 Web
 iOS ✔️ 
 Android ✔️ 
 Desktop App
 Mobile Web

**Version Number:** v1.0.50-2
**Expensify/Expensify Issue URL:** https://github.com/Expensify/Expensify/issues/164496

**Upwork job link:** https://www.upwork.com/jobs/~01b1a9d276904a90c7

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

- [x] -> Profile page
- [x] -> Password Page
- [x] -> Payments page
- [x] -> New group Page
- [x] -> Add Secondary login
- [x] -> IOU -> Request Money Page.
- [x] -> IOU -> Currency Selection Page
- [x] -> IOU -> Request Money -> Confirm Page
- [x] -> IOU -> Split Money -> Confirm Page
- [x] -> IOU -> Split Money -> Participant Page
 

### Fixed Issues

Fixes #3055

### Tests | QA Steps
1.Open E.cash on Android and IOS.
2. Check all the above listed Pages.
3. Fixed Button button(if any) should always float over the keyboard. 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### iOS

https://user-images.githubusercontent.com/24370807/120418006-710df300-c37d-11eb-8192-dd0910550ab7.mp4

#### Android

https://user-images.githubusercontent.com/24370807/119906801-898ca080-bf6c-11eb-9e41-71504e03e8f5.mp4

### CODE REVIEW

**Comment by parasharrajat** [comment]
Requesting review @shawnborton

File: src/pages/iou/IOUCurrencySelection.js [hunk 1]
```diff
@@ -163,69 +166,65 @@ class IOUCurrencySelection extends Component {
...
-                                )}
-                            />
+                            <View style={[styles.ph5, styles.pv3]}>
+                                <TextInputWithFocusStyles
+                                    styleFocusIn={[styles.textInputReversedFocus]}
+                                    ref={el => this.textInput = el}
+                                    style={[styles.textInput]}
+                                    value={this.state.searchValue}
+                                    onChangeText={this.changeSearchValue}
+                                    placeholder="Search"
```
Comment (Dal-Papa) [inline src/pages/iou/IOUCurrencySelection.js:183]: Shouldn't this get translated too ?

**Review by Dal-Papa — COMMENTED** [review]
Apart from my comment about i18n, this looks ok to me but I'd love @marcaaron's opinion on it.

File: src/pages/iou/IOUCurrencySelection.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/iou/IOUCurrencySelection.js:183]: Yeah, it should be. But It does not come under the scope of this PR. Anyways happy to do this change. Let me know, if I should do this change as well.

**Comment by shawnborton** [comment]
This looks good to me! I think we should standardize on the button styles for disabled buttons though. My thinking is that a disabled button should be 50% transparent, giving us something like this:
image

Thoughts?

**Comment by parasharrajat** [comment]
I am not sure about it. It totally depends on the Branding. I think we should raise this point on the Slack. @shawnborton

**Comment by shawnborton** [comment]
Well that is indeed the branding that we have designed, it just hasn't been implemented correctly. But I understand that this isn't technically part of the work we asked you to do, so we can open up a separate issue to address the button inconsistencies.

**Comment by michaelhaxhiu** [comment]
@shawnborton feel free to drop a comment in #expensify-open-source by saying something like `ISSUE: standardize button styles` and we can run it through the standard process to get it assigned!

**Comment by parasharrajat** [comment]
Updated. Marc is on leave till Monday. Maybe we can have it reviewed by someone else. Thanks.

File: src/pages/iou/IOUCurrencySelection.js (hunk 1 above)
Comment (Dal-Papa) [reply src/pages/iou/IOUCurrencySelection.js:183]: Could you please add it ? That way we don't forget about it.

**Comment by Dal-Papa** [comment]
@roryabraham mind giving this a quick look to be safe please ? 🙏

File: src/pages/iou/IOUCurrencySelection.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/iou/IOUCurrencySelection.js:183]: Pushing a commit right away.

Comment (parasharrajat) [reply src/pages/iou/IOUCurrencySelection.js]: @Dal-Papa Code pushed. Thanks.

**Review by Dal-Papa — DISMISSED** [review]
LGTM

File: src/components/FixedFooter.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+
+const propTypes = {
+    /** Children to wrap in FixedFooter. */
+    children: PropTypes.node.isRequired,
+
+    /** Styles to be assigned to Container */
+    style: PropTypes.arrayOf(PropTypes.object),
+
```
Comment (Jag96) [inline src/components/FixedFooter.js:12]: unnecessary space

File: src/libs/KeyboardAvoidingView/index.js [hunk 3]
```diff
@@ -14,18 +11,9 @@ const defaultProps = {
...
-        >
-            {children}
-        </KeyboardAvoidingViewComponent>
-    );
-}
+const KeyboardAvoidingView = props => props.children;
 
 KeyboardAvoidingView.propTypes = propTypes;
 KeyboardAvoidingView.defaultProps = defaultProps;
+KeyboardAvoidingView.displayName = 'KeyboardAvoidingView';
```
Comment (Jag96) [inline src/libs/KeyboardAvoidingView/index.js:18]: It looks like the index.ios.js counterpart doesn't have the displayName set, can we set it there as well?

**Comment by parasharrajat** [comment]
@roryabraham Are we good here? Thanks.

**Review by Jag96 — COMMENTED** [review]
Left a few notes but overall looks good

File: src/libs/KeyboardAvoidingView/index.js (hunk 3 above)
Comment (parasharrajat) [reply src/libs/KeyboardAvoidingView/index.js:18]: adding

**Comment by parasharrajat** [comment]
@Jag96 Done.

**Comment by roryabraham** [comment]
> I think we should standardize on the button styles for disabled buttons though. My thinking is that a disabled button should be 50% transparent, giving us something like this

Agree, but in addition we should standardize on a `cursor: not-allowed` style when hovering over any disabled button:

image

File: src/pages/iou/IOUModal.js [hunk 4]
```diff
@@ -305,42 +306,42 @@ class IOUModal extends Component {
...
-                                                this.navigateToNextStep();
-                                            }}
-                                            currencySelected={this.currencySelected}
-                                            reportID={this.props.route.params.reportID}
-                                            selectedCurrency={this.state.selectedCurrency}
-                                            hasMultipleParticipants={this.props.hasMultipleParticipants}
-                                            selectedAmount={this.state.amount}
-                                            navigation={this.props.navigation}
-                                        />
+                                    <IOUAmountPage
```
Comment (roryabraham) [inline src/pages/iou/IOUModal.js:309]: All this indentation was correct, can we reset this back the way it was?

**Review by roryabraham — CHANGES_REQUESTED** [review]
Overall this PR leaves me wondering if we can make some reusable components to make some of this stuff easier in the future, something like this (in psuedocode):

```jsx
const ViewWithFixedButtonFooter = props => (
    <>
        <ScrollView style={[styles.flex1, styles.w100]}>
            {this.props.children}
        </ScrollView>
        {this.props.shouldShowButton && (
            <FixedFooter>
                <Button
                    success={this.props.buttonSuccess}
                    onPress={this.props.onButtonPressed}
                    style={styles.w100}
                    text={this.props.buttonText}
                />
            </FixedFooter>
        )}
    </>
)

const PageWithFixedButtonFooter = props => (
    <ScreenWrapper>
        {({didScreenTransitionEnd}) => (
            <>
                <FullScreenLoadingIndicator visible={!didScreenTransitionEnd} />
                {didScreenTransitionEnd && (
                    <KeyboardAvoidingView>
                        {this.props.shouldShowHeader && (
                            <HeaderWithCloseButton />
                        )}
                        <ViewWithFixedButtonFooter>
                            {this.props.children}
                        </ViewWithFixedButtonFooter />
                    </KeyboardAvoidingView>
                )}
            </>
        )}
    </ScreenWrapper>
)
```

**Comment by parasharrajat** [comment]
Yeah. It does look that a component can be extracted but I don't think it would be that much value as it seems. Let me have a close look at it to see if we have a pattern here.

**Comment by michaelhaxhiu** [comment]
👋 heyo - is this on track to merge today?

**Comment by parasharrajat** [comment]
I am looking into extracting the code into common components but I feel like it requires some amount of time to carefully replace all the components. I can put a follow-up PR and I think we should merge this as it is working as it should be. But it's your call.

**Comment by michaelhaxhiu** [comment]
@parasharrajat thanks for the context. Let's have @roryabraham weigh in if he didn't already before making a decision. 

But my 2 cents is we could decide this based on scope. If the addition you are describing is going to inflate scope significantly, my vote is to open a new PR. But if it's a part of the current scope then I agree with holding and keeping it contained in 1 PR for simplicity.

**Comment by roryabraham** [comment]
Sorry for the delay here @parasharrajat, if you resolve the merge conflicts, retest, and it's all still working then I'd feel okay approving/merging.

In my experience, all this stuff with `KeyboardAvoidingView`, `ScrollView`, and now `FixedFooter` is always pretty challenging to get right on all different devices. It seems like you've done a good job getting everything working (which is what we hired you for 😁 ), so we can merge this! But it would be more valuable if we could extract some reusable components from this to make the pattern easier to repeat and get right in the future. Just :my_two_cents:

**Comment by parasharrajat** [comment]
@roryabraham I agree with the point to extract common components but currently, I feel a little overwhelmed. I would definitely look into abstracting them. I have updated the PR and let me know what would be the best step ahead.

**Review by roryabraham — APPROVED** [review]
No worries @parasharrajat! I'm sure you've got higher-priority stuff that you're assigned, and this is working. Let's just move on and work on new features and bug fixes. Maybe we can circle back on this if a similar issue arises in the future 😄

**Pull request merged**